docs/adr/0046-monorepo-split.md

```diff
@@ -0,0 +1,143 @@
+# Keep SubPlat in the Nx monorepo with targeted improvements
+
+* **Status:** proposed
+* **Deciders:** Julian Poyourow, Reino Muhl, Vijay Budhram
+* **Date:** 2025-08-18
+
+## Context and Problem Statement
+
+FxA (Mozilla Accounts) and SubPlat (Subscription Platform) share a long‑lived monorepo that migrated to Nx about two years ago. The monorepo contains:
+
+* **Older/Legacy yarn-workspace-based under `packages/*`** (e.g., `auth-server`, `admin-server`, the legacy `payments-server` UI), a mix of JavaScript and TypeScript in an older style.
+* **Modern Nx‑library based code under `apps/*` and `libs/*`**, including `apps/payments-next` (Next.js) and domain libraries:
+
+  * `libs/payments/*` (SubPlat)
+  * `libs/accounts/*` (FxA)
+  * `libs/shared/*` (shared CMS, logging, etc.)
+  * `libs/*` other
+
+Historically, SubPlat processed most payments, emails, and webhooks via `auth-server`. Over the last year, SubPlat moved payment processing and the frontend to `payments-next` (Next.js), increasing independence while still depending on `auth-server` for various functions. `admin-server` also spans both domains.
+
+SubPlat has considered moving to a separate repository for the maintenance and deployment of Subscription Platform-specific functionality.
+
+**Pain points that led to considering a split:**
+
+* Slow CI (often ≥ \~20 minutes).
+* Legacy TS+JS in `auth-server` with wide use of typecasts and older testing methodologies.
+* Coupled release process: FxA and SubPlat are forced to tag and deploy together.
+* Dependency upgrades are painful across the repo.
+* Consistency of code style between the two teams varies.
```

Comment:
Why?

Reply:
If I had to guess it's more architecture and structural patterns than pure style. These patterns often lend themselves to frameworks and languages the code is written in / on. Now that the subplat stack and the accounts stack have diverged, I feel like its inevitable the patterns and idioms we use in code will drift as well. 

In addition, the fact the two teams are kind of siloed, prevents us from developing a consistent style. Accounts doesn't code review subplat PRs and subplat doesn't code review accounts PRs. 

I'd love to hear Julian's thoughts on this though.